drivers/blk/sdmmc/src/main.rs

```diff
@@ -0,0 +1,279 @@
+#![no_std] // Don't link the standard library
+#![no_main] // Don't use the default entry point
+
+extern crate alloc;
+
+mod sddf_blk;
+
+use core::{
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll, RawWaker, RawWakerVTable, Waker},
+};
+
+use alloc::boxed::Box;
+use sddf_blk::{
+    blk_dequeue_req_helper, blk_enqueue_resp_helper, blk_queue_empty_req_helper,
+    blk_queue_full_resp_helper, blk_queue_init_helper, BlkOp, BlkRequest, BlkStatus,
+};
+use sdmmc_hal::meson_gx_mmc::MesonSdmmcRegisters;
+
+use sdmmc_protocol::sdmmc::{InterruptType, SdmmcHalError, SdmmcHardware, SdmmcProtocol};
+use sel4_microkit::{debug_print, debug_println, protection_domain, Channel, Handler, Infallible};
+
+const BLK_VIRTUALIZER: sel4_microkit::Channel = sel4_microkit::Channel::new(0);
+
+const INTERRUPT: sel4_microkit::Channel = sel4_microkit::Channel::new(1);
+
+const SDCARD_SECTOR_SIZE: u32 = 512;
+const SDDF_TRANSFER_SIZE: u32 = 4096;
+const SDDF_TO_REAL_SECTOR: u32 = SDDF_TRANSFER_SIZE / SDCARD_SECTOR_SIZE;
+
+const RETRY_CHANCE: u16 = 5;
```

Comment:
Can you explain this retries thing?

Reply:
When sdcard encounters an error, like time out, crc error or other kinds of error, the sdmmc driver should retry a few times before reporting back an error in case the error is only temporary. I think uboot does this as well.